File: backend/src/routes/person_info.py
"""
Person Info API 路由
人物信息相关的API端点
"""

import time
from fastapi import APIRouter, HTTPException, Query
from typing import Optional

from models.person_info import (
    PersonInfoCreate, PersonInfoUpdate, PersonInfoDelete,
    PersonInfoQuery, PersonInfoListResponse, PersonInfoCreateResponse,
    PersonInfoUpdateResponse, PersonInfoDeleteResponse
)
from services.person_info_service import person_info_service
from core.database_manager import database_manager
import logging

logger = logging.getLogger("HMML")

# 创建路由器
router = APIRouter(prefix="/database/person-info", tags=["人物信息"])


def create_success_response(data: Optional[dict] = None, message: str = "操作成功") -> dict:
    """创建成功响应"""
    return {
        "status": 200,
        "message": message,
        "data": data,
        "time": int(time.time() * 1000)
    }


def create_error_response(status: int, message: str) -> dict:
    """创建错误响应"""
    return {
        "status": status,
        "message": message,
        "time": int(time.time() * 1000)
    }


@router.get("/get", response_model=PersonInfoListResponse, summary="查询人物信息列表")
async def get_person_info_list(
    page: int = Query(1, ge=1, description="页码，从1开始"),
    pageSize: int = Query(10, ge=1, le=100, description="每页数量"),
    person_id: Optional[str] = Query(None, description="人物ID过滤"),
    platform: Optional[str] = Query(None, description="平台过滤"),
    user_id: Optional[str] = Query(None, description="用户ID过滤"),
    person_name: Optional[str] = Query(None, description="人物名称过滤")
):
    """
    查询人物信息列表
    
    Args:
        page: 页码，从1开始
        pageSize: 每页数量
        person_id: 人物ID过滤（可选）
        platform: 平台过滤（可选）
        user_id: 用户ID过滤（可选）
        person_name: 人物名称过滤（可选）
        
    Returns:
        人物信息列表响应
    """
    try:
        logger.debug(f"处理查询人物信息列表请求，页码: {page}, 每页: {pageSize}")
        
        # 构建查询参数
        query = PersonInfoQuery(
            page=page,
            pageSize=pageSize,
            person_id=person_id,
            platform=platform,
            user_id=user_id,
            person_name=person_name
        )
        
        # 调用服务层
        result = await person_info_service.get_person_info_list(query)
        
        # 转换为API响应格式
        response_data = {
            "items": [item.model_dump() for item in result.items],
            "totalPages": result.totalPages,
            "currentPage": result.currentPage,
            "pageSize": result.pageSize
        }
        
        logger.info(f"查询人物信息列表成功，共 {len(result.items)} 条记录")
        return create_success_response(response_data, "查询成功")
        
    except Exception as error:
        logger.error(f"查询人物信息列表失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "查询失败")
        )


@router.post("/insert", response_model=PersonInfoCreateResponse, summary="插入人物信息")
async def insert_person_info(request: PersonInfoCreate):
    """
    插入人物信息
    
    Args:
        request: 人物信息数据
        
    Returns:
        插入结果响应
    """
    try:
        logger.debug(f"处理插入人物信息请求: {request.person_name}")
        
        # 调用服务层
        new_id = await person_info_service.create_person_info(request)
        
        response_data = {"id": new_id}
        
        logger.info(f"插入人物信息成功，ID: {new_id}")
        return create_success_response(response_data, "插入成功")
        
    except Exception as error:
        logger.error(f"插入人物信息失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "插入失败")
        )


@router.post("/update", response_model=PersonInfoUpdateResponse, summary="更新人物信息")
async def update_person_info(request: PersonInfoUpdate):
    """
    更新人物信息
    
    Args:
        request: 人物信息数据（包含ID）
        
    Returns:
        更新结果响应
    """
    try:
        logger.debug(f"处理更新人物信息请求，ID: {request.id}")
        
        # 检查记录是否存在
        existing = await person_info_service.get_person_info_by_id(request.id)
        if not existing:
            logger.warning(f"更新失败，记录不存在，ID: {request.id}")
            raise HTTPException(
                status_code=404,
                detail=create_error_response(404, "记录不存在")
            )
        
        # 调用服务层
        success = await person_info_service.update_person_info(request)
        
        if success:
            logger.info(f"更新人物信息成功，ID: {request.id}")
            return create_success_response(message="更新成功")
        else:
            raise HTTPException(
                status_code=500,
                detail=create_error_response(500, "更新失败")
            )
        
    except HTTPException:
        raise
    except Exception as error:
        logger.error(f"更新人物信息失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "更新失败")
        )


@router.delete("/delete", response_model=PersonInfoDeleteResponse, summary="删除人物信息")
async def delete_person_info(request: PersonInfoDelete):
    """
    删除人物信息
    
    Args:
        request: 删除请求（包含ID）
        
    Returns:
        删除结果响应
    """
    try:
        logger.debug(f"处理删除人物信息请求，ID: {request.id}")
        
        # 检查记录是否存在
        existing = await person_info_service.get_person_info_by_id(request.id)
        if not existing:
            logger.warning(f"删除失败，记录不存在，ID: {request.id}")
            raise HTTPException(
                status_code=404,
                detail=create_error_response(404, "记录不存在")
            )
        
        # 调用服务层
        success = await person_info_service.delete_person_info(request.id)
        
        if success:
            logger.info(f"删除人物信息成功，ID: {request.id}")
            return create_success_response(message="删除成功")
        else:
            raise HTTPException(
                status_code=500,
                detail=create_error_response(500, "删除失败")
            )
        
    except HTTPException:
        raise
    except Exception as error:
        logger.error(f"删除人物信息失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "删除失败")
        )


@router.get("/get/{person_id}", summary="根据ID获取单个人物信息")
async def get_person_info_by_id(person_id: int):
    """
    根据ID获取单个人物信息
    
    Args:
        person_id: 人物信息ID
        
    Returns:
        单个人物信息响应
    """
    try:
        logger.debug(f"处理根据ID查询人物信息请求，ID: {person_id}")
        
        # 调用服务层
        person_info = await person_info_service.get_person_info_by_id(person_id)
        
        if not person_info:
            logger.warning(f"未找到人物信息，ID: {person_id}")
            raise HTTPException(
                status_code=404,
                detail=create_error_response(404, "记录不存在")
            )
        
        logger.info(f"查询人物信息成功，ID: {person_id}")
        return create_success_response(person_info.model_dump(), "查询成功")
        
    except HTTPException:
        raise
    except Exception as error:
        logger.error(f"查询人物信息失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "查询失败")
        )


@router.put("/update/{person_id}", summary="更新人物信息（RESTful风格）")
async def update_person_info_restful(person_id: int, request: PersonInfoCreate):
    """
    更新人物信息（RESTful风格）
    
    Args:
        person_id: 人物信息ID
        request: 人物信息数据
        
    Returns:
        更新结果响应
    """
    try:
        logger.debug(f"处理RESTful更新人物信息请求，ID: {person_id}")
        
        # 检查记录是否存在
        existing = await person_info_service.get_person_info_by_id(person_id)
        if not existing:
            logger.warning(f"更新失败，记录不存在，ID: {person_id}")
            raise HTTPException(
                status_code=404,
                detail=create_error_response(404, "记录不存在")
            )
        
        # 构建更新对象
        update_request = PersonInfoUpdate(id=person_id, **request.model_dump())
        
        # 调用服务层
        success = await person_info_service.update_person_info(update_request)
        
        if success:
            logger.info(f"更新人物信息成功，ID: {person_id}")
            return create_success_response(message="更新成功")
        else:
            raise HTTPException(
                status_code=500,
                detail=create_error_response(500, "更新失败")
            )
        
    except HTTPException:
        raise
    except Exception as error:
        logger.error(f"更新人物信息失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "更新失败")
        )


@router.delete("/delete/{person_id}", summary="删除人物信息（RESTful风格）")
async def delete_person_info_restful(person_id: int):
    """
    删除人物信息（RESTful风格）
    
    Args:
        person_id: 人物信息ID
        
    Returns:
        删除结果响应
    """
    try:
        logger.debug(f"处理RESTful删除人物信息请求，ID: {person_id}")
        
        # 检查记录是否存在
        existing = await person_info_service.get_person_info_by_id(person_id)
        if not existing:
            logger.warning(f"删除失败，记录不存在，ID: {person_id}")
            raise HTTPException(
                status_code=404,
                detail=create_error_response(404, "记录不存在")
            )
        
        # 调用服务层
        success = await person_info_service.delete_person_info(person_id)
        
        if success:
            logger.info(f"删除人物信息成功，ID: {person_id}")
            return create_success_response(message="删除成功")
        else:
            raise HTTPException(
                status_code=500,
                detail=create_error_response(500, "删除失败")
            )
        
    except HTTPException:
        raise
    except Exception as error:
        logger.error(f"删除人物信息失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "删除失败")
        )


@router.get("/stats", summary="获取人物信息统计")
async def get_person_info_stats():
    """
    获取人物信息统计信息
    
    Returns:
        统计信息响应
    """
    try:
        logger.debug("处理获取人物信息统计请求")
        
        # 获取数据库操作器
        operator = database_manager.get_maibot_operator()
        if not operator:
            raise RuntimeError("数据库连接不可用")
        
        # 基础统计
        total_result = operator.execute_raw_sql("SELECT COUNT(*) as total FROM person_info")
        total = total_result.fetchone()['total'] if total_result else 0
        
        # 按平台统计
        platform_result = operator.execute_raw_sql("SELECT platform, COUNT(*) as count FROM person_info GROUP BY platform")
        by_platform = {row['platform']: row['count'] for row in platform_result.fetchall()}
        
        # 计算平均认知次数
        avg_know_times_result = operator.execute_raw_sql("SELECT AVG(CAST(know_times AS FLOAT)) as avg_know_times FROM person_info WHERE know_times IS NOT NULL")
        avg_know_times = float(avg_know_times_result.fetchone()['avg_know_times'] or 0)
        
        # 计算平均友好度
        avg_friendly_result = operator.execute_raw_sql("SELECT AVG(CAST(friendly_value AS FLOAT)) as avg_friendly FROM person_info WHERE friendly_value IS NOT NULL")
        avg_friendly_value = float(avg_friendly_result.fetchone()['avg_friendly'] or 0)
        
        # 总认知次数
        total_know_times_result = operator.execute_raw_sql("SELECT SUM(CAST(know_times AS FLOAT)) as total_know_times FROM person_info WHERE know_times IS NOT NULL")
        total_know_times = float(total_know_times_result.fetchone()['total_know_times'] or 0)
        
        # 按友好度分组统计
        friendly_stats_result = operator.execute_raw_sql("""
            SELECT 
                CASE 
                    WHEN friendly_value IS NULL THEN 'unknown'
                    WHEN CAST(friendly_value AS INTEGER) <= 2 THEN 'low'
                    WHEN CAST(friendly_value AS INTEGER) <= 5 THEN 'medium'
                    WHEN CAST(friendly_value AS INTEGER) <= 8 THEN 'high'
                    ELSE 'very_high'
                END as level,
                COUNT(*) as count
            FROM person_info 
            GROUP BY level
        """)
        by_friendly_value = {row['level']: row['count'] for row in friendly_stats_result.fetchall()}
        
        # 按态度分组统计  
        attitude_stats_result = operator.execute_raw_sql("""
            SELECT 
                CASE 
                    WHEN attitude_to_me IS NULL THEN 'unknown'
                    WHEN CAST(attitude_to_me AS INTEGER) <= 2 THEN 'negative'
                    WHEN CAST(attitude_to_me AS INTEGER) <= 5 THEN 'neutral'
                    WHEN CAST(attitude_to_me AS INTEGER) <= 8 THEN 'positive'
                    ELSE 'very_positive'
                END as level,
                COUNT(*) as count
            FROM person_info 
            GROUP BY level
        """)
        by_attitude_to_me = {row['level']: row['count'] for row in attitude_stats_result.fetchall()}
        
        # 按粗鲁度分组统计
        rudeness_stats_result = operator.execute_raw_sql("""
            SELECT 
                CASE 
                    WHEN rudeness IS NULL THEN 'unknown'
                    WHEN CAST(rudeness AS INTEGER) <= 2 THEN 'low'
                    WHEN CAST(rudeness AS INTEGER) <= 5 THEN 'medium'
                    WHEN CAST(rudeness AS INTEGER) <= 8 THEN 'high'
                    ELSE 'very_high'
                END as level,
                COUNT(*) as count
            FROM person_info 
            GROUP BY level
        """)
        by_rudeness = {row['level']: row['count'] for row in rudeness_stats_result.fetchall()}
        
        # 按神经质程度分组统计
        neuroticism_stats_result = operator.execute_raw_sql("""
            SELECT 
                CASE 
                    WHEN neuroticism IS NULL THEN 'unknown'
                    WHEN CAST(neuroticism AS INTEGER) <= 2 THEN 'low'
                    WHEN CAST(neuroticism AS INTEGER) <= 5 THEN 'medium'
                    WHEN CAST(neuroticism AS INTEGER) <= 8 THEN 'high'
                    ELSE 'very_high'
                END as level,
                COUNT(*) as count
            FROM person_info 
            GROUP BY level
        """)
        by_neuroticism = {row['level']: row['count'] for row in neuroticism_stats_result.fetchall()}
        
        # 按尽责程度分组统计
        conscientiousness_stats_result = operator.execute_raw_sql("""
            SELECT 
                CASE 
                    WHEN conscientiousness IS NULL THEN 'unknown'
                    WHEN CAST(conscientiousness AS INTEGER) <= 2 THEN 'low'
                    WHEN CAST(conscientiousness AS INTEGER) <= 5 THEN 'medium'
                    WHEN CAST(conscientiousness AS INTEGER) <= 8 THEN 'high'
                    ELSE 'very_high'
                END as level,
                COUNT(*) as count
            FROM person_info 
            GROUP BY level
        """)
        by_conscientiousness = {row['level']: row['count'] for row in conscientiousness_stats_result.fetchall()}
        
        # 按喜爱程度分组统计
        likeness_stats_result = operator.execute_raw_sql("""
            SELECT 
                CASE 
                    WHEN likeness IS NULL THEN 'unknown'
                    WHEN CAST(likeness AS INTEGER) <= 2 THEN 'low'
                    WHEN CAST(likeness AS INTEGER) <= 5 THEN 'medium'
                    WHEN CAST(likeness AS INTEGER) <= 8 THEN 'high'
                    ELSE 'very_high'
                END as level,
                COUNT(*) as count
            FROM person_info 
            GROUP BY level
        """)
        by_likeness = {row['level']: row['count'] for row in likeness_stats_result.fetchall()}
        
        # 最近活跃用户（7天内有互动的）
        recent_active_result = operator.execute_raw_sql("""
            SELECT COUNT(*) as recent_count 
            FROM person_info 
            WHERE last_know IS NOT NULL 
            AND last_know > ?
        """, (time.time() - 7 * 24 * 3600,))  # 7天前的时间戳
        recent_active = recent_active_result.fetchone()['recent_count'] if recent_active_result else 0
        
        # 认知次数最多的前5个人
        top_persons_result = operator.execute_raw_sql("""
            SELECT id, person_name, know_times 
            FROM person_info 
            WHERE know_times IS NOT NULL 
            ORDER BY CAST(know_times AS FLOAT) DESC 
            LIMIT 5
        """)
        top_persons = []
        for row in top_persons_result.fetchall():
            top_persons.append({
                "id": row['id'],
                "person_name": row['person_name'],
                "know_times": float(row['know_times']) if row['know_times'] else 0
            })
        
        # 构建完整统计数据
        stats_data = {
            "total": total,
            "byPlatform": by_platform,
            "byFriendlyValue": by_friendly_value,
            "byAttitudeToMe": by_attitude_to_me,
            "byRudeness": by_rudeness,
            "byNeuroticism": by_neuroticism,
            "byConscientiousness": by_conscientiousness,
            "byLikeness": by_likeness,
            "avgKnowTimes": avg_know_times,
            "avgFriendlyValue": avg_friendly_value,
            "totalKnowTimes": total_know_times,
            "recentActive": recent_active,
            "topPersons": top_persons
        }
        
        logger.info("获取人物信息统计成功")
        return create_success_response(stats_data, "获取统计信息成功")
        
    except Exception as error:
        logger.error(f"获取人物信息统计失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "获取统计信息失败")
        )


@router.get("/platforms", summary="获取可用平台列表")
async def get_available_platforms():
    """
    获取可用平台列表
    
    Returns:
        平台列表响应
    """
    try:
        logger.debug("处理获取可用平台列表请求")
        
        # 获取数据库操作器
        operator = database_manager.get_maibot_operator()
        if not operator:
            raise RuntimeError("数据库连接不可用")
        
        # 查询平台列表
        result = operator.execute_raw_sql("SELECT DISTINCT platform FROM person_info ORDER BY platform")
        
        platforms = [{"name": row['platform'], "count": 0} for row in result.fetchall()]
        
        # 获取每个平台的数量
        for platform in platforms:
            count_result = operator.execute_raw_sql("SELECT COUNT(*) as count FROM person_info WHERE platform = ?", (platform['name'],))
            platform['count'] = count_result.fetchone()['count'] if count_result else 0
        
        logger.info("获取可用平台列表成功")
        return create_success_response(platforms, "获取平台列表成功")
        
    except Exception as error:
        logger.error(f"获取可用平台列表失败: {error}")
        raise HTTPException(
            status_code=500,
            detail=create_error_response(500, "获取平台列表失败")
        )

File: backend/old/src/routes/personInfo.ts
/**
 * PersonInfo表操作路由
 * 提供PersonInfo的RESTful API接口
 */

import Router from '@koa/router';
import { Context, Next } from 'koa';
import { PersonInfoService } from '../services/personInfoService';
import { 
  PersonInfoInsertData, 
  PersonInfoUpdateData, 
  PersonInfoPaginationParams,
  PersonInfoFilterOptions,
  Platform
} from '../types/personInfo';
import { ValidationError, NotFoundError } from '../middleware/errorHandler';

const router = new Router();

/**
 * GET /database/person-info/get
 * 分页查询PersonInfo
 * 查询参数：
 * - page: 页码 (默认1)
 * - pageSize: 每页记录数 (默认20)
 * - orderBy: 排序字段 (默认id)
 * - orderDir: 排序方向 ASC|DESC (默认ASC)
 * - person_id: 过滤人物ID
 * - person_name: 过滤人物名称 (模糊匹配)
 * - platform: 过滤平台
 * - user_id: 过滤用户ID
 * - nickname: 过滤昵称 (模糊匹配)
 * - is_known: 过滤是否认识 (0或1)
 * - points: 过滤认知点 (模糊匹配)
 * - attitude_to_me: 过滤对自己的态度 (模糊匹配)
 * - rudeness: 过滤粗鲁度 (模糊匹配)
 * - neuroticism: 过滤神经质程度 (模糊匹配)
 * - conscientiousness: 过滤尽责程度 (模糊匹配)
 * - likeness: 过滤喜爱程度 (模糊匹配)
 * - minKnowTimes: 最小认知次数
 * - maxKnowTimes: 最大认知次数
 * - minFriendlyValue: 最小友好度
 * - maxFriendlyValue: 最大友好度
 * - knowSinceStart: 认识起始时间戳
 * - knowSinceEnd: 认识结束时间戳
 * - lastKnowStart: 最后认知起始时间戳
 * - lastKnowEnd: 最后认知结束时间戳
 */
router.get('/person-info/get', async (ctx: Context, next: Next) => {
  try {
    const query = ctx.query as Record<string, string>;
    
    // 解析分页参数
    const page = Math.max(1, parseInt(query.page) || 1);
    const pageSize = Math.min(100, Math.max(1, parseInt(query.pageSize) || 20));
    const orderBy = query.orderBy || 'id';
    const orderDir = query.orderDir === 'DESC' ? 'DESC' : 'ASC';

    // 构建过滤条件
    const filter: PersonInfoFilterOptions = {};
    
    if (query.person_id) filter.person_id = query.person_id;
    if (query.person_name) filter.person_name = query.person_name;
    if (query.platform) filter.platform = query.platform;
    if (query.user_id) filter.user_id = query.user_id;
    if (query.nickname) filter.nickname = query.nickname;
    if (query.is_known !== undefined) {
      const isKnown = parseInt(query.is_known);
      if (isKnown === 0 || isKnown === 1) {
        filter.is_known = isKnown;
      }
    }
    if (query.points) filter.points = query.points;
    if (query.attitude_to_me) filter.attitude_to_me = query.attitude_to_me;
    if (query.rudeness) filter.rudeness = query.rudeness;
    if (query.neuroticism) filter.neuroticism = query.neuroticism;
    if (query.conscientiousness) filter.conscientiousness = query.conscientiousness;
    if (query.likeness) filter.likeness = query.likeness;
    
    if (query.minKnowTimes !== undefined) {
      filter.minKnowTimes = parseInt(query.minKnowTimes);
    }
    if (query.maxKnowTimes !== undefined) {
      filter.maxKnowTimes = parseInt(query.maxKnowTimes);
    }
    if (query.minFriendlyValue !== undefined) {
      filter.minFriendlyValue = parseFloat(query.minFriendlyValue);
    }
    if (query.maxFriendlyValue !== undefined) {
      filter.maxFriendlyValue = parseFloat(query.maxFriendlyValue);
    }
    if (query.knowSinceStart !== undefined) {
      filter.knowSinceStart = parseInt(query.knowSinceStart);
    }
    if (query.knowSinceEnd !== undefined) {
      filter.knowSinceEnd = parseInt(query.knowSinceEnd);
    }
    if (query.lastKnowStart !== undefined) {
      filter.lastKnowStart = parseInt(query.lastKnowStart);
    }
    if (query.lastKnowEnd !== undefined) {
      filter.lastKnowEnd = parseInt(query.lastKnowEnd);
    }

    const params: PersonInfoPaginationParams = {
      page,
      pageSize,
      orderBy,
      orderDir,
      filter: Object.keys(filter).length > 0 ? filter : undefined
    };

    const result = await PersonInfoService.getPersonInfos(params);
    
    ctx.body = {
      status: 200,
      message: '查询PersonInfo成功',
      data: result,
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * GET /database/person-info/get/:id
 * 根据ID查询单个PersonInfo
 */
router.get('/person-info/get/:id', async (ctx: Context, next: Next) => {
  try {
    const id = parseInt(ctx.params.id);
    
    if (isNaN(id) || id < 1) {
      throw new ValidationError('ID必须是大于0的数字');
    }

    const result = await PersonInfoService.getPersonInfoById(id);
    
    if (!result) {
      throw new NotFoundError(`未找到ID为${id}的PersonInfo记录`);
    }

    ctx.body = {
      status: 200,
      message: '查询PersonInfo成功',
      data: result,
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * POST /database/person-info/insert
 * 新增PersonInfo
 * 请求体：PersonInfoInsertData
 */
router.post('/person-info/insert', async (ctx: Context, next: Next) => {
  try {
    const data: PersonInfoInsertData = (ctx.request as any).body;
    
    if (!data || typeof data !== 'object') {
      throw new ValidationError('请求体必须是有效的JSON对象');
    }

    const insertId = await PersonInfoService.insertPersonInfo(data);
    
    ctx.body = {
      status: 200,
      message: 'PersonInfo创建成功',
      data: {
        id: insertId,
        ...data
      },
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * PUT /database/person-info/update/:id
 * 更新PersonInfo
 * 请求体：PersonInfoUpdateData (不包含id)
 */
router.put('/person-info/update/:id', async (ctx: Context, next: Next) => {
  try {
    const id = parseInt(ctx.params.id);
    const data: Partial<PersonInfoInsertData> = (ctx.request as any).body;
    
    if (isNaN(id) || id < 1) {
      throw new ValidationError('ID必须是大于0的数字');
    }

    if (!data || typeof data !== 'object') {
      throw new ValidationError('请求体必须是有效的JSON对象');
    }

    // 构建更新数据
    const updateData: PersonInfoUpdateData = { id, ...data };
    
    await PersonInfoService.updatePersonInfo(updateData);
    
    ctx.body = {
      status: 200,
      message: 'PersonInfo更新成功',
      data: updateData,
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * DELETE /database/person-info/delete/:id
 * 删除PersonInfo
 */
router.delete('/person-info/delete/:id', async (ctx: Context, next: Next) => {
  try {
    const id = parseInt(ctx.params.id);
    
    if (isNaN(id) || id < 1) {
      throw new ValidationError('ID必须是大于0的数字');
    }

    await PersonInfoService.deletePersonInfo(id);
    
    ctx.body = {
      status: 200,
      message: 'PersonInfo删除成功',
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * GET /database/person-info/by-person/:personId
 * 根据person_id查询PersonInfo列表
 */
router.get('/person-info/by-person/:personId', async (ctx: Context, next: Next) => {
  try {
    const personId = ctx.params.personId;
    
    if (!personId) {
      throw new ValidationError('person_id不能为空');
    }

    const result = await PersonInfoService.getPersonInfoByPersonId(personId);
    
    ctx.body = {
      status: 200,
      message: `查询person_id为${personId}的PersonInfo成功`,
      data: result,
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * GET /database/person-info/by-platform/:platform/user/:userId
 * 根据平台和用户ID查询PersonInfo列表
 */
router.get('/person-info/by-platform/:platform/user/:userId', async (ctx: Context, next: Next) => {
  try {
    const platform = ctx.params.platform;
    const userId = ctx.params.userId;
    
    if (!platform || !userId) {
      throw new ValidationError('platform和userId不能为空');
    }

    const result = await PersonInfoService.getPersonInfoByPlatformUser(platform, userId);
    
    ctx.body = {
      status: 200,
      message: `查询平台${platform}上用户${userId}的PersonInfo成功`,
      data: result,
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * GET /database/person-info/stats
 * 获取PersonInfo统计信息
 */
router.get('/person-info/stats', async (ctx: Context, next: Next) => {
  try {
    const result = await PersonInfoService.getPersonInfoStats();
    
    ctx.body = {
      status: 200,
      message: '获取PersonInfo统计信息成功',
      data: result,
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * GET /database/person-info/search
 * 搜索PersonInfo (支持按姓名、昵称、认知点、对自己的态度搜索)
 * 查询参数：
 * - keyword: 搜索关键字 (必需)
 * - limit: 限制数量 (默认20，最大100)
 */
router.get('/person-info/search', async (ctx: Context, next: Next) => {
  try {
    const keyword = ctx.query.keyword as string;
    const limit = Math.min(100, Math.max(1, parseInt(ctx.query.limit as string) || 20));
    
    if (!keyword || keyword.trim().length === 0) {
      throw new ValidationError('搜索关键字不能为空');
    }

    const result = await PersonInfoService.searchPersonInfos(keyword.trim(), limit);
    
    ctx.body = {
      status: 200,
      message: `搜索PersonInfo成功，找到${result.length}条记录`,
      data: result,
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

/**
 * GET /database/person-info/platforms
 * 获取可用平台列表
 */
router.get('/person-info/platforms', async (ctx: Context, next: Next) => {
  try {
    const platforms = Object.values(Platform);
    
    ctx.body = {
      status: 200,
      message: '获取平台列表成功',
      data: platforms,
      time: Date.now()
    };

  } catch (error) {
    throw error;
  }
});

export { router as personInfoRouter };
